From ea693750fe45f250303008e0fe32b41ebe9a3cca Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jean-Andr=C3=A9=20Santoni?= <jean.andre.santoni@gmail.com>
Date: Fri, 28 Dec 2018 17:00:07 +0700
Subject: [PATCH 1/5] Fix cross compilation of libX11

I need to cross compile this package for one of my projects.

`makekeys` has to be built for the host architecture, not the target architecture.

See more detailed explanations here about why this change make sense
https://github.com/hanshuebner/buildroot/blob/master/package/x11r7/xlib_libX11/xlib_libX11.mk#L24-L29

This change will also help in building LibreELEC from another architecture than x86_64.
---
 packages/x11/lib/libX11/package.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/packages/x11/lib/libX11/package.mk b/packages/x11/lib/libX11/package.mk
index 2548a07f592..3a92765304b 100644
--- a/packages/x11/lib/libX11/package.mk
+++ b/packages/x11/lib/libX11/package.mk
@@ -30,3 +30,9 @@ PKG_CONFIGURE_OPTS_TARGET="--disable-loadable-i18n \
                            --disable-ipv6 \
                            --without-launchd \
                            --without-lint"
+
+make_target() {
+  echo '' > $PKG_BUILD/src/util/Makefile.am
+  $HOST_CC $PKG_BUILD/src/util/makekeys.c -o $PKG_BUILD/.$TARGET_NAME/src/util/makekeys
+  make V=1 CC=$CC AR=$AR LD=$LD CFLAGS="$TARGET_CFLAGS"
+}

From 792d56b54b9168aa5a853b6ae2a1488688cde1d5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jean-Andr=C3=A9=20Santoni?= <jean.andre.santoni@gmail.com>
Date: Sat, 29 Dec 2018 22:32:20 +0700
Subject: [PATCH 2/5] Address comments

---
 packages/x11/lib/libX11/package.mk | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/packages/x11/lib/libX11/package.mk b/packages/x11/lib/libX11/package.mk
index 3a92765304b..c292d492c2b 100644
--- a/packages/x11/lib/libX11/package.mk
+++ b/packages/x11/lib/libX11/package.mk
@@ -31,8 +31,11 @@ PKG_CONFIGURE_OPTS_TARGET="--disable-loadable-i18n \
                            --without-launchd \
                            --without-lint"
 
-make_target() {
+pre_make_target() {
   echo '' > $PKG_BUILD/src/util/Makefile.am
   $HOST_CC $PKG_BUILD/src/util/makekeys.c -o $PKG_BUILD/.$TARGET_NAME/src/util/makekeys
+}
+
+make_target() {
   make V=1 CC=$CC AR=$AR LD=$LD CFLAGS="$TARGET_CFLAGS"
 }

From 49439671fd8f04255d45e56138976b06b077b6b9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jean-Andr=C3=A9=20Santoni?= <jean.andre.santoni@gmail.com>
Date: Sun, 30 Dec 2018 13:43:08 +0700
Subject: [PATCH 3/5] Fix the build

---
 packages/x11/lib/libX11/package.mk | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/packages/x11/lib/libX11/package.mk b/packages/x11/lib/libX11/package.mk
index c292d492c2b..3a92765304b 100644
--- a/packages/x11/lib/libX11/package.mk
+++ b/packages/x11/lib/libX11/package.mk
@@ -31,11 +31,8 @@ PKG_CONFIGURE_OPTS_TARGET="--disable-loadable-i18n \
                            --without-launchd \
                            --without-lint"
 
-pre_make_target() {
+make_target() {
   echo '' > $PKG_BUILD/src/util/Makefile.am
   $HOST_CC $PKG_BUILD/src/util/makekeys.c -o $PKG_BUILD/.$TARGET_NAME/src/util/makekeys
-}
-
-make_target() {
   make V=1 CC=$CC AR=$AR LD=$LD CFLAGS="$TARGET_CFLAGS"
 }

From 683016998fdaa318b513641c5f260485dca0c82b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jean-Andr=C3=A9=20Santoni?= <jean.andre.santoni@gmail.com>
Date: Sun, 30 Dec 2018 15:16:14 +0700
Subject: [PATCH 4/5] Update package.mk

---
 packages/x11/lib/libX11/package.mk | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/packages/x11/lib/libX11/package.mk b/packages/x11/lib/libX11/package.mk
index 3a92765304b..62b279e6266 100644
--- a/packages/x11/lib/libX11/package.mk
+++ b/packages/x11/lib/libX11/package.mk
@@ -8,7 +8,7 @@ PKG_SHA256="4d3890db2ba225ba8c55ca63c6409c1ebb078a2806de59fb16342768ae63435d"
 PKG_LICENSE="OSS"
 PKG_SITE="http://www.x.org/"
 PKG_URL="http://xorg.freedesktop.org/archive/individual/lib/$PKG_NAME-$PKG_VERSION.tar.bz2"
-PKG_DEPENDS_TARGET="toolchain util-macros xtrans libXau libxcb"
+PKG_DEPENDS_TARGET="toolchain util-macros xtrans libXau libxcb xorgproto:host"
 PKG_LONGDESC="LibX11 is the main X11 library containing all the client-side code to access the X11 windowing system."
 PKG_TOOLCHAIN="autotools"
 
@@ -31,8 +31,11 @@ PKG_CONFIGURE_OPTS_TARGET="--disable-loadable-i18n \
                            --without-launchd \
                            --without-lint"
 
-make_target() {
+pre_make_target() {
   echo '' > $PKG_BUILD/src/util/Makefile.am
-  $HOST_CC $PKG_BUILD/src/util/makekeys.c -o $PKG_BUILD/.$TARGET_NAME/src/util/makekeys
-  make V=1 CC=$CC AR=$AR LD=$LD CFLAGS="$TARGET_CFLAGS"
+  $HOST_CC $HOST_CFLAGS -I$PKG_BUILD/include $PKG_BUILD/src/util/makekeys.c -o $PKG_BUILD/.$TARGET_NAME/src/util/makekeys
+}
+
+make_target() {
+  make V=1 CC=$CC LD=$LD CFLAGS="$TARGET_CFLAGS"
 }

From 0cf35a5d68aba0e4dedbf9d8e0d2d241babcfec2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jean-Andr=C3=A9=20Santoni?= <jean.andre.santoni@gmail.com>
Date: Mon, 31 Dec 2018 01:54:10 +0700
Subject: [PATCH 5/5] Update package.mk

---
 packages/x11/lib/libX11/package.mk | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/packages/x11/lib/libX11/package.mk b/packages/x11/lib/libX11/package.mk
index 62b279e6266..01d81cc2ec7 100644
--- a/packages/x11/lib/libX11/package.mk
+++ b/packages/x11/lib/libX11/package.mk
@@ -35,7 +35,3 @@ pre_make_target() {
   echo '' > $PKG_BUILD/src/util/Makefile.am
   $HOST_CC $HOST_CFLAGS -I$PKG_BUILD/include $PKG_BUILD/src/util/makekeys.c -o $PKG_BUILD/.$TARGET_NAME/src/util/makekeys
 }
-
-make_target() {
-  make V=1 CC=$CC LD=$LD CFLAGS="$TARGET_CFLAGS"
-}
